types/abci.go

```diff
@@ -23,6 +23,13 @@ type ProcessProposalHandler func(Context, *abci.ProcessProposalRequest) (*abci.P
 // PrepareProposalHandler defines a function type alias for preparing a proposal
 type PrepareProposalHandler func(Context, *abci.PrepareProposalRequest) (*abci.PrepareProposalResponse, error)
 
+// CheckTxHandler defines a function type alias for executing logic before transactions are executed.
+// `RunTx` is a function type alias for executing logic before transactions are executed.
+type CheckTxHandler func(RunTx, *abci.CheckTxRequest) (*abci.CheckTxResponse, error)
+
+// RunTx defines a function type alias for executing logic before transactions are executed.
```

Comment:
Can we just have this inline in CheckTxHandler? we don't expect anyone else to use RunTx, and this is a subset of runtx